plan/property/stats_info.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package property
+
+import (
+	"fmt"
+
+	"github.com/pingcap/tidb/statistics"
+)
+
+// StatsInfo stores the basic information of statistics for the plan's output. It is used for cost estimation.
+type StatsInfo struct {
```

Comment:
I prefer put it and the things about `deriveStatsInfo` into a single sub package of plan.

Reply:
`deriveStatsInfo` is part of the optimization step, maybe it's better to put these functions in the old engine package?

Reply:
`statsInfo` can be treated as property of plan node to some extent, `deriveStats` mainly does range extraction, stats computation, and branch prune, this part of optimization should be in the new planner as well IMHO?

Reply:
Yes, and the situation is more complicated in the new planner because we maintain the `StatsInfo` for a group rather than a expression.